resume-buddy/src/main/java/com/google/sps/Match.java

```diff
@@ -0,0 +1,55 @@
+package com.google.sps;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import java.util.List;
+
+/** Class that matches reviewers to reviewees */
+public class Match {
```

Comment:
I think this functionality would work better as public static functions instead of private members. When I think about the use case, I think writing something like: 

```
Match match = new Match();
```

And it feels odd that there isn't really a use for the created object after construction. There is also a difficulty in testing this API.

Another way this could be done is by having `getNonMatchedUsers` return an object containing the reviewees and reviewers that are unmatched. Then the `match` function could take two lists as arguments and returns the matches. Then a third function could do the database update.

It's quite possible the split is not necessary for testing but I haven't been able to find a mock datastore service class yet.

Reply:
I modified the code and now matching can be called following way: 

`List<Entity> reviewees = Match.getNotMatchedUsers("Reviewee");` 
`List<Entity> reviewers = Match.getNotMatchedUsers("Reviewer");`
`Match.match(reviewees, reviewers);`

I'll add test cases to test match method.